src/sprightly.ts

```diff
@@ -11,33 +11,41 @@ interface Options {
   cache?: boolean
 }
 
-function parse(file: string, entryPoint: string, data: Data, options: Options) {
-  const parsedFile = file.replaceAll(
+class SprightlyError extends Error {
+  constructor(message: string) {
+    super(message)
+  }
+}
+
+function parse(
+  fileContent: string,
+  filePath: string,
+  data: Data,
+  options: Options,
+) {
+  const parsedFile = fileContent.replaceAll(
     /\{\{(>?)(.*?)\}\}/g,
     (_, isComponent: string, reference: string) => {
       reference = reference.trim()
       if (isComponent) {
-        // if last segement of path is a file
         // we go back one step to get the directory of the file
-        if (path.extname(entryPoint)) {
-          entryPoint = path.normalize(path.join(entryPoint, ".."))
-        }
+        const fileDirectory = path.normalize(path.join(filePath, ".."))
 
         const componentAbsolutePath = path.resolve(
```

Comment:
Here you can `.join` because you can assume that the resulting path is an absolute one. You want to make this assumption because its part of the `parse` function API to deal with absolute paths, so if this assertion fails, the code will **fail fast**,